certbot/filelock.py

```diff
@@ -0,0 +1,324 @@
+"""Implements file locks for locking files and directories in UNIX and Windows."""
+import errno
+import logging
+import os
+import atexit
+try:
+    import fcntl  # pylint: disable=import-error
+    POSIX_MODE = True
+except ImportError:
+    import msvcrt  # pylint: disable=import-error
+    POSIX_MODE = False
+
+from certbot import errors
+from acme.magic_typing import List, Optional  # pylint: disable=unused-import, no-name-in-module
+
+logger = logging.getLogger(__name__)
+
+
+# Handling exit
+# ~~~~~~~~~~~~~
+
+
+_INITIAL_PID = os.getpid()
+_LOCKS = []  # type: List[FileLock]
+
+
+def _release_all_locks():
+    """Release all locks acquired by FileLock."""
+    if _INITIAL_PID == os.getpid():
+        for lock in _LOCKS:
```

Comment:
While it's not a big deal, I still think we should consider alternatives for duplicating the code here. Do you have any ideas?

Reply:
Removed along atexit logic removal.